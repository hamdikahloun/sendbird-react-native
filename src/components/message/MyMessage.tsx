import React, {memo} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {UserMessage} from 'sendbird';

type MassageProps = {
  message: UserMessage;
};
export const MyMessageMemorizd = ({message}: MassageProps) => {
  return (
    <View style={styles.container}>
      <Text style={styles.message}>{message.message}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 10,
    backgroundColor: '#6100FF',
    marginBottom: 16,
    padding: 16,
    borderRadius: 15,
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignSelf: 'flex-end',
  },
  message: {
    color: 'white',
  },
});

/**
 * Message - This UI component displays an individual preview item for each message in a list
 *
 * See:
 *
 * [BaseMessage](https://sendbird.github.io/core-sdk-javascript/module-model_baseMessage-BaseMessage.html)
 * @example
 *
 * <Message message={message} />
 */
export const MyMessage = memo(MyMessageMemorizd);
