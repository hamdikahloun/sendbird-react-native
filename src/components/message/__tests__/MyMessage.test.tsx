import React from 'react';
import {render} from '@/../testing';
import {MyMessage} from '../MyMessage';

describe('MyMessage', () => {
  it('should render correctly', async () => {
    const message: any = {message: ''};
    const screen = render(<MyMessage message={message} />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
