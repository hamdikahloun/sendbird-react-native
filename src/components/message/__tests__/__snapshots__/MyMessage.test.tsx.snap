// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMessage should render correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-end",
      "backgroundColor": "#6100FF",
      "borderRadius": 15,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginBottom": 16,
      "padding": 16,
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    
  </Text>
</View>
`;
